lib/filesystem/CFilesystemLoader.cpp

```diff
@@ -26,8 +26,9 @@ CFilesystemLoader::CFilesystemLoader(std::string _mountPoint, bfs::path baseDire
 std::unique_ptr<CInputStream> CFilesystemLoader::load(const ResourceID & resourceName) const
 {
 	assert(fileList.count(resourceName));
-
-	return make_unique<CFileInputStream>(baseDirectory / fileList.at(resourceName));
+	boost::filesystem::path file = baseDirectory / fileList.at(resourceName);
```

Comment:
Again not really your code problem, but that file has:
```namespace bfs = boost::filesystem;```
So you might use ```bfs::path``` instead.

Or you can just use ```auto``` because in context of that file is plain enough what type it is.